sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/action/ReflectiveActionRouter.scala

```diff
@@ -50,23 +51,33 @@ class ReflectiveActionRouter[A <: Action](
     val inputTypeUrl = message.payload().asInstanceOf[ScalaPbAny].typeUrl
     val methodInvoker = commandHandler.lookupInvoker(inputTypeUrl)
 
-    methodInvoker match {
-      case Some(invoker) =>
-        inputTypeUrl match {
-          case ProtobufEmptyTypeUrl =>
-            invoker
-              .invoke(action)
-              .asInstanceOf[Action.Effect[_]]
-          case _ =>
-            invoker
-              .invoke(action, invocationContext)
-              .asInstanceOf[Action.Effect[_]]
-        }
-      case None if ignoreUnknown => ActionEffectImpl.Builder.ignore()
-      case None =>
-        throw new NoSuchElementException(
-          s"Couldn't find any method with input type [$inputTypeUrl] in Action [$action].")
-
+    // lookup ComponentClient
+    val componentClients =
+      Reflect
+        .lookupComponentClientField(action.getClass, action)
```

Comment:
Small bug found when trying this with the sample..

```suggestion
        .lookupComponentClientField(action)
```